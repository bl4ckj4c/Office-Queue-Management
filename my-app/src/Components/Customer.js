import { Col, Row, Container } from 'react-bootstrap';
import { Button, Card, Modal } from 'react-bootstrap';
import { useState, useEffect } from 'react';

import API from '../API.js';
import "./customer.css";
import icon1 from './icons/1.jpg';
import icon2 from './icons/2.jpg';
import icon3 from './icons/3.jpg';
import icon4 from './icons/4.jpg';

function Customer(props) {

    const [ticketNum, setTicketNum] = useState(-1);
    const [selectedService, setSelectedService] = useState(null); 
    const [modalShow, setModalShow] = useState(false);
    const closeModal = () => setModalShow(false);
  
    useEffect(() => {
      const getTicket = async () => {
        const ticket = await API.getTicket(selectedService);
        setTicketNum(ticket);
        setModalShow(true);
      };


	if(selectedService){
        getTicket();
        setSelectedService(null);
	}
	

    }, [selectedService]);
  
    return (
        <Container>
            <ShowTicketModal show={modalShow} handleClose={()=>setModalShow(false)} message={ticketNum}/>

            <Row className="justify-content-center mt-3">GET YOUR TICKET</Row>
            <Row className="justify-content-center mt-3">
                <Col xs lg="4">
                    <Card>
                        <Card.Img variant="top" src={icon1} />
                        <Card.Body>
                            <Card.Title>BILL</Card.Title>
                            <Button variant="primary" onClick={() => setSelectedService(1)}>Add me to queue</Button>
                        </Card.Body>
                    </Card>
                </Col>

                <Col xs lg="4">
                    <Card>
                        <Card.Img variant="top" src={icon2} />
                        <Card.Body>
                            <Card.Title>POST</Card.Title>
                            <Button variant="primary" onClick={() => setSelectedService(2)}>Add me to queue</Button>
                        </Card.Body>
                    </Card>
                </Col>
            </Row>
            <Row className="justify-content-center mt-2 mb-2">
                <Col xs lg="4">
                    <Card>
                        <Card.Img variant="top" src={icon3}/>
                        <Card.Body>
                            <Card.Title>BOX</Card.Title>
                            <Button variant="primary" onClick={() => setSelectedService(3)}>Add me to queue</Button>
                        </Card.Body>
                    </Card>
                </Col>
                <Col xs lg="4">
                    <Card>
                        <Card.Img variant="top" src={icon4}/>
                        <Card.Body>
                            <Card.Title>WITHDRAWAL</Card.Title>
                            <Button variant="primary" onClick={() => setSelectedService(4)}>Add me to queue</Button>
                        </Card.Body>
                    </Card>
                </Col>
            </Row>
        </Container>

    );
};

function ShowTicketModal(props) {

    return (
      <Modal
        show={props.show}
        onHide={props.handleClose}
        size="md"
        aria-labelledby="contained-modal-title-vcenter"
        centered>
        <Modal.Header className="text-center font-weight-bold" closeButton onClick={props.handleClose}>Dear customer, here's your ticket number: </Modal.Header>
        <Modal.Body className="display-1 text-center font-weight-bold">{props.message.number}</Modal.Body>
      </Modal>
    );
}
  

export default Customer;

