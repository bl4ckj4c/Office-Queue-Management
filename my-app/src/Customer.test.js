import { render, screen } from '@testing-library/react';
import Customer from './Components/Customer';

test('Snapshot', () => {
  const tree = render(
    <Customer> </Customer>
  );
  expect(tree).toMatchSnapshot();
});

test.todo('Test that button trigger the right service');