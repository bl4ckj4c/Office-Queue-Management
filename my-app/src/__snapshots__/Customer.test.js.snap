// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="justify-content-center mt-3 text row"
        >
          GET YOUR TICKET
        </div>
        <div
          class="justify-content-center mt-3 row"
        >
          <div
            class="col-lg-4 col"
          >
            <div
              class="card"
            >
              <img
                class="card-img-top"
                src="1.jpg"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  SERVICE 1
                </div>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Add me to queue
                </button>
              </div>
            </div>
          </div>
          <div
            class="col-lg-4 col"
          >
            <div
              class="card"
            >
              <img
                class="card-img-top"
                src="2.jpg"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  SERVICE 2
                </div>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Add me to queue
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="justify-content-center mt-2 mb-2 row"
        >
          <div
            class="col-lg-4 col"
          >
            <div
              class="card"
            >
              <img
                class="card-img-top"
                src="3.jpg"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  SERVICE 3
                </div>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Add me to queue
                </button>
              </div>
            </div>
          </div>
          <div
            class="col-lg-4 col"
          >
            <div
              class="card"
            >
              <img
                class="card-img-top"
                src="4.jpg"
              />
              <div
                class="card-body"
              >
                <div
                  class="card-title h5"
                >
                  SERVICE 4
                </div>
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Add me to queue
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="justify-content-center mt-3 text row"
      >
        GET YOUR TICKET
      </div>
      <div
        class="justify-content-center mt-3 row"
      >
        <div
          class="col-lg-4 col"
        >
          <div
            class="card"
          >
            <img
              class="card-img-top"
              src="1.jpg"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                SERVICE 1
              </div>
              <button
                class="btn btn-primary"
                type="button"
              >
                Add me to queue
              </button>
            </div>
          </div>
        </div>
        <div
          class="col-lg-4 col"
        >
          <div
            class="card"
          >
            <img
              class="card-img-top"
              src="2.jpg"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                SERVICE 2
              </div>
              <button
                class="btn btn-primary"
                type="button"
              >
                Add me to queue
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="justify-content-center mt-2 mb-2 row"
      >
        <div
          class="col-lg-4 col"
        >
          <div
            class="card"
          >
            <img
              class="card-img-top"
              src="3.jpg"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                SERVICE 3
              </div>
              <button
                class="btn btn-primary"
                type="button"
              >
                Add me to queue
              </button>
            </div>
          </div>
        </div>
        <div
          class="col-lg-4 col"
        >
          <div
            class="card"
          >
            <img
              class="card-img-top"
              src="4.jpg"
            />
            <div
              class="card-body"
            >
              <div
                class="card-title h5"
              >
                SERVICE 4
              </div>
              <button
                class="btn btn-primary"
                type="button"
              >
                Add me to queue
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
